code/lib/types/src/modules/composedStory.ts

```diff
@@ -30,40 +39,60 @@ export type ComposedStoryPlayFn<TRenderer extends Renderer = Renderer, TArgs = A
   context: ComposedStoryPlayContext<TRenderer, TArgs>
 ) => Promise<void> | void;
 
-export type PreparedStoryFn<TRenderer extends Renderer = Renderer, TArgs = Args> = AnnotatedStoryFn<
-  TRenderer,
-  TArgs
-> & { play: ComposedStoryPlayFn<TRenderer, TArgs>; args: TArgs; id: StoryId };
+/**
+ * A loosely annotated story function, used internally by composeStory
+ */
+export type LooselyAnnotatedStoryFn<TRenderer extends Renderer = Renderer, TArgs = Args> = (
+  args?: TArgs
+) => (TRenderer & {
+  T: TArgs;
+})['storyResult'];
+
+/**
+ * A story that got recomposed for portable stories, containing all the necessary data to be rendered in external environments
+ */
+export type ComposedStoryFn<
+  TRenderer extends Renderer = Renderer,
+  TArgs = Args
+> = LooselyAnnotatedStoryFn<TRenderer, TArgs> & {
+  play: ComposedStoryPlayFn<TRenderer, TArgs>;
+  args: TArgs;
+  id: StoryId;
+  storyName: string;
+  parameters: Parameters;
+};
 
-export type ComposedStory<TRenderer extends Renderer = Renderer, TArgs = Args> =
+/**
+ * Type that matches whether the story is a CSF2 story or a CSF3 story, used for assertion and inference
+ */
+export type StoryLike<TRenderer extends Renderer = Renderer, TArgs = Args> =
```

Comment:
Why did we call this `ComposedStory` before? Seems backwards, isn't it something we are about to compose?

`StoryLike` -- I'm not sure this is right, it's not "story like" it *is* a story (annotations). Is there a type in here that expresses this? Maybe `StoryAnnotationsOrFn`? https://github.com/ComponentDriven/csf/blob/4c735fea4f0c9605b93497238303cb4ab9304727/src/story.ts#L394-L396

Reply:
I would go for `StoryAnnotationsOrFn` indeed

Reply:
Makes total sense, I didn't know this type already existed!